script.js

```diff
@@ -5,10 +5,69 @@ const classNames = {
   TODO_DELETE: 'todo-delete',
 }
 
+let itemCount = 0
+let uncheckedItemsCount = 0
+let toDos = []
+
 const list = document.getElementById('todo-list')
 const itemCountSpan = document.getElementById('item-count')
 const uncheckedCountSpan = document.getElementById('unchecked-count')
 
+
 function newTodo() {
-  alert('New TODO button clicked!')
+	itemCount++
+	itemCountSpan.innerHTML = itemCount
+	const toDo = createToDo()
+	toDos.push(toDo)
+	list.appendChild(toDo)
 }
+
+function createToDo(){
+	const li = document.createElement('li')
+	li.className = classNames.TODO_ITEM
+	
+	const text = document.createElement('input')
+	text.setAttribute('value', 'TODO')
+	text.className = classNames.TODO_TEXT
+
+	const checkBox = document.createElement('input')
+	checkBox.className = classNames.TODO_CHECKBOX
+	checkBox.type = 'checkBox'
+	checkBox.onchange = updateIsChecked
+
+	const deleteButton = document.createElement('button')
+	deleteButton.className = classNames.TODO_DELETE
+	deleteButton.innerHTML = 'delete'
+	deleteButton.onclick = deleteToDo
+	deleteButton.referanceToLi = li
+
+	uncheckedItemsCount++
+	uncheckedCountSpan.innerHTML = uncheckedItemsCount
+	
+	li.appendChild(checkBox)
+	li.appendChild(text)
+	li.appendChild(deleteButton)
+	return li
+}
+function deleteToDo() {
+	itemCount--
+	itemCountSpan.innerHTML = itemCount
+	if (!this.checked) {
+		uncheckedItemsCount--
+		uncheckedCountSpan.innerHTML = uncheckedItemsCount
+	}
+	toDos = toDos.filter(todo => this.referanceToLi !== todo)
+	list.innerHTML = ''
+	toDos.forEach(todo => list.appendChild(todo))
```

Comment:
Can you try to delete a specific item instead of append all items again?

Reply:
resolved as well